distribution/packages/build.gradle

```diff
@@ -213,7 +213,7 @@ Closure commonPackageConfig(String type, boolean jdk, String architecture) {
     configurationFile '/etc/opensearch/jvm.options'
     configurationFile '/etc/opensearch/log4j2.properties'
     from("${packagingFiles}") {
-      dirMode 02750
+      dirMode 0750
```

Comment:
It looks to be a good change from the security perspective, but definitely lack the context, specifically https://github.com/elastic/elasticsearch/pull/37500 seemed to purposely change `0750` to  `02750`

Reply:
Well, we can see in that PR that the 750 -> 2750 is related to the removal of `chmod g+s`:
https://github.com/elastic/elasticsearch/pull/37500/files#diff-02fe6840ff0d98ed2b587ea2f5e7f928b0fbd0846a987e1b7278f4bd09768758L99

Looking at the file history, we can track the addition of this SGID bit in this commit:
https://github.com/rjernst/elasticsearch/commit/a5c18f15b252747bc8c62372bc8e07a36572e441

While working on https://github.com/opensearch-project/opensearch-build/pull/3898, I **wanted** to change the files ownership in `/etc/opensearch` to `root:opensearch` but the service does not initially start because the opensearch user needs to have write permission in this directory to create the keystore.  As a consequence:
1. I decided to keep files ownership to the opensearch user (meaning that the service has permission to rewrite it's own configuration which is never a good sign), assuming that we may improve this point later;
2. The purpose of the introduction of the SGID bit is moot, and we can remove it.